spec/abci/abci++_basic_concepts.md

```diff
@@ -438,3 +441,16 @@ part of a decided block, the `Code` does not influence consensus.
 
 When CometBFT receives a `QueryResponse` with a non-zero `Code`, this code is
 returned directly to the client that initiated the query.
+
+### ABCI methods `error` return
+
+The `error` return, the second object returned in a ABCI method e.g., `(*abcitypes.[Method_Name]Response, error)`, is utilized in situations
+involving unrecoverable errors.
+
+All ABCI methods return errors. An error in any of these methods represents a critical issue that CometBFT
+has no reasonable way to handle. If there is an error in one of these methods, the application must crash
+to ensure that an operator safely handles the error.
+
+In such cases, it is recommended that the application be terminated to avoid any further unintended consequences.
```

Comment:
```suggestion
to ensure that an operator safely handles the error: the application must be terminated to avoid any further unintended consequences.

As a result, upon detecting an non-recoverable error condition, the application has the choice of either (a) crashing itself (e.g., a`panic` in the code detecting the unrecoverable condition), or (b) returning an error as the second return value in the ABCI method that detects the error condition, known that CometBFT will panic upon receiving that error. The choice between (a) and (b) is up to the application -- both are equivalent -- and depends on whether an application (e.g. running in a different process that CometBFT) prefers CometBFT to crash first.
```